tox_extension_messages.c

```diff
@@ -220,6 +330,18 @@ void tox_extension_messages_handle_message_finish(
 
 	tox_extension_copy_in_message_data(parsed_packet, incoming_message);
 
+	struct FriendSendingInvalidSize *friend_dropping =
```

Comment:
Something about this feels off

1. Why are we copying data in if we know we're dropping it?
2. This is duplicated with the section above. Seems like it's happening unconditionally. I think if friend_data->sending_invalid we should just unconditionally clear the message and return.